### PR TITLE
Add py-imageio for python3

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Add new package py-imageio.

Don't merge until concretizer can choose dependencies for python 2 vs 3. (Redundant dependencies can be added as a workaround if people need this.)

Note:  ```https://pypi.io/packages/source/I/imageio/imageio-2.3.0.tar.gz``` does not work

### CODE REVIEW

**Comment by adamjstewart** [comment]
> Note: https://pypi.io/packages/source/I/imageio/imageio-2.3.0.tar.gz does not work

Try https://pypi.io/packages/source/i/imageio/imageio-2.3.0.tar.gz

**Comment by s-sajid-ali** [comment]
Done. Changed the checksum to match the pypi version this time!

File: var/spack/repos/builtin/packages/py-imageio/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    # Fix for python 2 if needed.
+    depends_on('py-numpy',  type=('build','run'))
+    depends_on('py-pillow', type=('build', 'run'))
+    depends_on('ffmpeg',    type='run')
+
+
+    def build_args(self, spec, prefix):
+        args = []
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-imageio/package.py]: You can remove this function if you aren't going to use it.

File: var/spack/repos/builtin/packages/py-imageio/package.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    url      = "https://pypi.io/packages/source/i/imageio/imageio-2.3.0.tar.gz"
+
+    version('2.3.0', '4722c4e1c366748abcb18729881cffb8')
+
+    #TODO: Add variants for plugins, and optional dependencies
+
+    # Fix for python 2 if needed.
+    depends_on('py-numpy',  type=('build','run'))
+    depends_on('py-pillow', type=('build', 'run'))
+    depends_on('ffmpeg',    type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-imageio/package.py]: Missing the following dependencies:
```python
depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
depends_on('py-setuptools', type='build')
```

**Comment by s-sajid-ali** [comment]
The setup.py from the github repo has this 
```
 install_requires = [
        'numpy',
        'pillow',
        'enum34; python_version < "3.4"',
        'futures; python_version < "3.2"'
],
```
Since the last two depend on the version of python and spack can't pick dependencies based on python 2 vs 3, I stated that if someone is using python2 and wants this, they can add a depends on py-enum34 and py-futures as a stopgap until the concretizer is fixed.

**Comment by adamjstewart** [comment]
I see. Well the latest tarball doesn't have these restrictions, so I think we can ignore this problem for now. Hopefully the concretizer will be updated before someone needs to add a new release of `py-imageio`.

**Comment by s-sajid-ali** [comment]
I'm getting a strange error when I try to activate it. What causes this ?
```
[sajid@xrmlite intel-18.0.3]$ spack find -ldf py-imageio
==> 1 installed packages.
-- linux-centos7-x86_64 / intel@18.0.3 --------------------------
u2nin6h    py-imageio@2.3.0%intel
2o6dsmy        ^ffmpeg@3.2.4%intel
ik46rji            ^yasm@1.3.0%intel
eu4bndh        ^py-numpy@1.14.3%intel
igylo3k            ^intel-mkl@2018.2.199%intel
gorhtot            ^python@3.6.5%intel
akk4pld                ^bzip2@1.0.6%intel
4krmbbu                ^gdbm@1.14.1%intel
c7bu4ts                    ^readline@7.0%intel
rpcqo55                        ^ncurses@6.1%intel
kxxpbno                ^openssl@1.0.2n%intel
vonxam4                    ^zlib@1.2.11%intel
t3b5otl                ^sqlite@3.23.1%intel
zo6tm3o        ^py-pillow@5.1.0%intel
37tkd4t            ^freetype@2.7.1%intel
tkydo2r                ^libpng@1.6.34%intel
ub3r732            ^lcms@2.9%intel
2v3pjm4                ^libjpeg-turbo@1.5.90%intel
nnft4ld                ^libtiff@4.0.9%intel
kgbcgub                    ^xz@5.2.4%intel

[sajid@xrmlite intel-18.0.3]$ spack activate py-imageio@2.3.0%intel
==> Error: Spec mfndl3jhvdmmxv7lvheusdti7fg7refa not found in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/py-imageio-2.3.0-mfndl3jhvdmmxv7lvheusdti7fg7refa```

It worked fine with ffmpeg@4.0.1, but I re-built it because that ffmpeg didn't work and now imageio won't load.

**Comment by s-sajid-ali** [comment]
Even if I force it to activate the correct version by DAG, it won't work. 
```
[sajid@xrmlite intel-18.0.3]$ spack activate /u2nin6h
==> Error: Spec mfndl3jhvdmmxv7lvheusdti7fg7refa not found in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/py-imageio-2.3.0-mfndl3jhvdmmxv7lvheusdti7fg7refa
```
Why is spack looking for the wrong dag ?

**Comment by healther** [comment]
> ==> Error: Spec mfndl3jhvdmmxv7lvheusdti7fg7refa not found in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/py-imageio-2.3.0-mfndl3jhvdmmxv7lvheusdti7fg7refa

This looks like a out-of-date database. Try running `spack reindex` or `spack clean -a && spack reindex`

**Comment by s-sajid-ali** [comment]
As per @adamjstewart , I tried ```module load ``` and it works. 

Clean and re-index and still the same error : 
```
[sajid@xrmlite spack]$ spack clean -a && spack reindex
==> Removing all temporary build stages
==> Removing cached downloads
==> Removing cached information on repositories
==> Removing python cache files
[sajid@xrmlite spack]$ spack activate py-imageio@2.3.0%intel@18.0.3
==> Error: Spec mfndl3jhvdmmxv7lvheusdti7fg7refa not found in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/py-imageio-2.3.0-mfndl3jhvdmmxv7lvheusdti7fg7refa
```

**Pull request merged**